### PR TITLE
Create LiveGoogleSheetView

### PR DESCRIPTION
**Pull request by RynhardPietersen**
## Product Description

Adds a new view to export side-bar to allow for user interfacing with gsheets.

## Technical Summary

[This](https://dimagi-dev.atlassian.net/browse/SAAS-13100) ticket was created in order for us to add in a new view in the same style as was done with OData.

## Feature Flag

Google Sheets

### Safety story

Local testing. This is also available to test on staging.

### QA Plan

QA will be done when the entire Google Sheets function is complete

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by RynhardPietersen** [comment]
In [this pr](https://github.com/dimagi/commcare-hq/pull/31310) I mention that I will add in code to fix the hard coded projectlimit for google sheets. Unfortunately that pr is yet to be merged and I was unable to add in that code as of yet. Once it has been merged I will merge master into this branch and add in that code.

File: corehq/apps/export/static/export/js/models.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
  *
  */
 
+const { contains } = require("jquery");
```
Comment (biyeun) [inline corehq/apps/export/static/export/js/models.js:7]: curious what this was for?

File: corehq/apps/export/templates/export/partials/export_list_controller.html [hunk 2]
```diff
@@ -29,10 +29,15 @@
         {% endblocktrans %}
       </a>
     </p>
-    {% if not odata_feeds_over_limit %}
+    {% if not feeds_over_limit %}
       {% include 'export/partials/export_list_create_export_modal.html' %}
     {% else %}
-      {% include 'export/partials/odata_feed_limit_reached_modal.html' %}
+      {% if isOData %}
+        {% include 'export/partials/odata_feed_limit_reached_modal.html' %}
```
Comment (biyeun) [inline corehq/apps/export/templates/export/partials/export_list_controller.html:36]: i think the URLs to these partials might need to be updated?

File: corehq/apps/export/views/list.py [hunk 3]
```diff
@@ -237,8 +242,18 @@ def fmt_export_data(self, export):
...
             'odataUrl': self._get_odata_url(export),
             'additionalODataUrls': self._get_additional_odata_urls(export),
+            'gsheetUrl': self._get_google_sheet_url(export),
         }
 
+    def _get_google_sheet_url(self, export):
+        try:
+            live_google_sheet_schedule = LiveGoogleSheetSchedule.objects.get(export_config_id=export)
+            google_sheet_id = live_google_sheet_schedule.google_sheet_id
+            google_sheet_url = "https://spreadsheets.google.com/feeds/worksheets/{}".format(google_sheet_id)
```
Comment (biyeun) [inline corehq/apps/export/views/list.py:252]: nit: i think an `f` string would be more readable here

File: corehq/apps/export/views/list.py [hunk 4]
```diff
@@ -895,6 +930,13 @@ def submit_app_data_drilldown_form(request, domain):
         else:
             export_tag = create_form.cleaned_data['form']
             cls = CreateODataFormFeedView
+    elif is_live_google_sheet:
+        if create_form.cleaned_data['model_type'] == "case":
+            export_tag = create_form.cleaned_data['case_type']
+            cls = CreateGoogleSheetCaseView
+        else:
+            export_tag = create_form.cleaned_data['form']
+            cls - CreateGoogleSheetFormView
```
Comment (biyeun) [inline corehq/apps/export/views/list.py:939]: this looks like a syntax error? should be `=`

File: corehq/apps/export/templates/export/partials/export_list_controller.html (hunk 2 above)
Comment (biyeun) [reply corehq/apps/export/templates/export/partials/export_list_controller.html:36]: also need to ensure that they get cleaned up / removed if they are no longer being used

File: corehq/apps/export/views/list.py [hunk 5]
```diff
@@ -1047,6 +1089,134 @@ def page_context(self):
...
+            model_type_choices.append(('form', _('Form')))
+        form.fields['model_type'].choices = model_type_choices
+
+        return form
+
+    @property
+    @memoized
+    def live_google_sheets_limit(self):
+        try:
+            domain_object = ProjectLimit.objects.get(domain=self.domain)
```
Comment (biyeun) [inline corehq/apps/export/views/list.py:1155]: this needs to be filtered by the gsheet `limit_type`. Also i think `domain_object` doesn't make sense as the variable name here? maybe `project_limit` instead?

File: corehq/apps/export/views/list.py [hunk 6]
```diff
@@ -1047,6 +1089,134 @@ def page_context(self):
...
+            return domain_object.limit_value
+        except ProjectLimit.DoesNotExist:
+            return settings.DEFAULT_GSHEET_LIMIT
+
+    @property
+    def create_export_form_title(self):
+        return _("Select Model Type")
+
+    def _should_appear_in_list(self, export):
+        test = IntegrationFormat.is_integration_format(export['export_format'])
```
Comment (biyeun) [inline corehq/apps/export/views/list.py:1165]: i think you want to actually see if the integration format is for live google sheets, not just if it is an integration format?

File: corehq/apps/export/views/list.py [hunk 7]
```diff
@@ -1047,6 +1089,134 @@ def page_context(self):
...
+    def dispatch(self, request, *args, **kwargs):
+        return super().dispatch(request, *args, **kwargs)
+
+    # TODO insert link to confluence page
+    @property
+    def lead_text(self):
+        return format_html(_("""
+        Integrate your live CommCare data with Google Sheets.
+        <a href=""
+       id="js-gsheet-track-learn-more"
```
Comment (biyeun) [inline corehq/apps/export/views/list.py:1201]: nit: usually attributes are indented so that they are all in line with the first attribute used (`href`)

**Review by biyeun — CHANGES_REQUESTED** [review]
great work so far! so happy to see this working end-to-end 🥳 

left a few comments for changes needing to be addressed

File: corehq/apps/export/static/export/js/models.js (hunk 1 above)
Comment (RynhardPietersen) [reply corehq/apps/export/static/export/js/models.js:7]: I'm not entirely sure...

File: corehq/apps/export/templates/export/partials/export_list_controller.html (hunk 2 above)
Comment (biyeun) [reply corehq/apps/export/templates/export/partials/export_list_controller.html:36]: @RynhardPietersen bump on this one

File: corehq/apps/export/static/export/js/models.js [hunk 8]
```diff
@@ -302,7 +302,8 @@ hqDefine('export/js/models', [
     };
 
     ExportInstance.prototype.getFormatOptionValues = function () {
-        return _.filter(constants.EXPORT_FORMATS, function (format) {
+        const availableExportFormats = Object.assign({}, constants.EXPORT_FORMATS, constants.INTEGRATED_EXPORT_FORMATS);
```
Comment (biyeun) [inline corehq/apps/export/static/export/js/models.js:305]: did you test these new changes on a regular export to ensure that google sheets doesn't show up in the dropdown? @RynhardPietersen

**Comment by AddisonDunn** [comment]
Hey @RynhardPietersen, I just merged in a branch that had been used to merge our conflicts on staging. There was only a small conflict for a single line in settings.py. I think I did this correctly but let me know if you notice any problems!

**Comment by snopoke** [comment]
Closing old PRs, reopen if needed.

**Pull request closed without merging**